index.bs

```diff
@@ -1701,6 +1712,41 @@ a [=moment=] |sourceTime|, and a [=duration=] |eventReportWindow|:
     1. Set |lastEnd| to |lastEnd| + |deadline|.
 1. Return |windows|.
 
+To <dfn>parse report windows</dfn> given an [=ordered map=] |map|, a [=moment=] |sourceTime|,
+and a [=list=] of [=event-level report windows=] |default|:
+
+1. Let |defaultEndDuration| be the [=duration from=] |sourceTime| and the [=event-level report window/end=] of the last element of |default|.
+1. If |map|["`event_report_windows`"] does not [=map/exist=], return |default|.
+1. Let |values| be |map|["`event_report_windows`"].
+1. If |values| is not a [=map=], return error.
+1. Let |startDuration| be 0 seconds.
+1. If |values|["`start_time`"] [=map/exists=]:
+    1. Set |start| to the result of applying the <a spec="html">rules for parsing integers</a> with |values|["`start_time`"].
+    1. If |start| is an error, return error.
+    1. If |start| is less than 0, return error.
+    1. Set |startDuration| to |start| seconds.
+    1. If |startDuration| is greater than |defaultEndDuration|, return error.
+1. If |values|["`end_times`"] does not [=map/exist=], return error.
+1. Let |endDurations| be |values|["`end_times`"].
+1. If the [=list/size=] of |endDurations| is greater than 5, return error.
+1. Let |windows| be an [=list/is empty=] [=list=].
+1. [=list/iterate|For each=] |elt| of |endDurations|:
+    1. Let |end| be the result of applying the <a spec="html">rules for parsing integers</a> to |elt|.
```

Comment:
Likewise, we may want to reuse one of our existing parsing algorithms. We could consider refactoring https://wicg.github.io/attribution-reporting-api/#parse-an-expiry in a separate change.

Reply:
I thought about reusing parse-an-expiry but it doesn't really fit the use-case. We could consider a more generic clamping algorithm but I think we can do that in a follow-up.